app/models/user/block_domain.rb

```diff
@@ -1,6 +1,7 @@
 class User::BlockDomain < ApplicationRecord
-  def self.blocked?(user)
-    domain = Mail::Address.new(user.email).domain
+  def self.blocked?(user_or_email)
+    email = user_or_email.is_a?(User) ? user_or_email.email : user_or_email
+    domain = Mail::Address.new(email).domain
```

Comment:
Maybe this method signature could be `user: nil, email: nil`, to make the pass in more explicit?

See `app/commands/donations/payment_intent/handle_invoice_failure.rb` for an example of this pattern.